src/client_ext.rs

```diff
@@ -0,0 +1,87 @@
+use crate::{Client, RequestBuilder};
+use std::sync::Arc;
+/// Extension trait that adds http request methods
+///
+/// Blanket implementation provided for all `http_client::HttpClient`s
+pub trait ClientExt {
+    /// Construct a new surf Client
+    fn client(&self) -> Client;
+
+    /// Builds a `CONNECT` request.
+    fn connect(&self, path: &str) -> RequestBuilder {
+        self.client().connect(path)
+    }
+
+    /// Builds a `DELETE` request.
+    fn delete(&self, path: &str) -> RequestBuilder {
+        self.client().delete(path)
+    }
+
+    /// Builds a `GET` request.
+    fn get(&self, path: &str) -> RequestBuilder {
+        self.client().get(path)
+    }
+
+    /// Builds a `HEAD` request.
+    fn head(&self, path: &str) -> RequestBuilder {
+        self.client().head(path)
+    }
+
+    /// Builds an `OPTIONS` request.
+    fn options(&self, path: &str) -> RequestBuilder {
+        self.client().options(path)
+    }
+
+    /// Builds a `PATCH` request.
+    fn patch(&self, path: &str) -> RequestBuilder {
+        self.client().patch(path)
+    }
+
+    /// Builds a `POST` request.
+    fn post(&self, path: &str) -> RequestBuilder {
+        self.client().post(path)
+    }
+
+    /// Builds a `PUT` request.
+    fn put(&self, path: &str) -> RequestBuilder {
+        self.client().put(path)
+    }
+
+    /// Builds a `TRACE` request.
+    fn trace(&self, path: &str) -> RequestBuilder {
+        self.client().trace(path)
+    }
+}
+
+impl<HC: http_client::HttpClient + Clone> ClientExt for HC {
+    fn client(&self) -> Client {
+        Client::with_http_client(Arc::new(self.clone()))
```

Comment:
It looks like this often may make something like `Arc<Arc<dyn HttpClient>>`. Maybe the signature of `Client::with_http_client()` needs to be improved.